src/sagemaker/amazon/hyperparameter.py

```diff
@@ -98,8 +100,14 @@ def serialize_all(obj):
         """
         if "_hyperparameters" not in dir(obj):
             return {}
-        return {
-            k: json.dumps(v) if isinstance(v, list) else str(v)
-            for k, v in obj._hyperparameters.items()
-            if v is not None
-        }
+        hps = {}
+        for k, v in obj._hyperparameters.items():
```

Comment:
how this change is related to enable the `source_dir` parameterization?

Reply:
Good call.

Without this fix, the existing unit test will fail: `test_training_step_with_algorithm_base`

The `source_dir` and `entry_point` should be added into the hyperparameter, that's why this fix is required here/